xarray/tests/test_weighted.py

```diff
@@ -49,6 +49,29 @@ def test_weighted_weights_nan_raises_dask(as_dataset, weights):
         weighted.sum().load()
 
 
+@requires_cftime
+@requires_dask
+@pytest.mark.parametrize("time_chunks", (1, 5))
+@pytest.mark.parametrize("resample_spec", ("1AS", "5AS", "10AS"))
+def test_weighted_lazy_resample(time_chunks, resample_spec):
+    # https://github.com/pydata/xarray/issues/4625
+
+    # simple customized weighted mean function
+    def mean_func(ds):
+        return ds.weighted(ds.weights).mean("time")
+
+    # example dataset
+    t = xr.cftime_range(start="2000", periods=1000, freq="1AS")
```

Comment:
can we use fewer periods? That should speed things up a lot

```suggestion
    t = xr.cftime_range(start="2000", periods=20, freq="1AS")
```

Reply:
Oh yea of course